File: ios/MassiveRPG/Domain/Model/CharacterClass.swift
//
//  CharacterClass.swift
//  MassiveRPG-Prod
//
//  Created by Onur Var on 10.04.2024.
//

enum CharacterClass: String, Codable {
    case Fighter = "fighter"
    case Cleric = "cleric"
    case Rogue = "rogue"
    case Wizard = "wizard"
    case Bard = "bard"

    var title: String {
        self.rawValue.uppercased()
    }

    static var `default`: CharacterClass = .Fighter
    static var list: [CharacterClass] {
        return [.Fighter, .Cleric, .Rogue, .Wizard, .Bard]
    }
}

File: android/app/src/main/java/com/onurvar/massiverpg/domain/model/CharacterClass.kt
package com.onurvar.massiverpg.domain.model

enum class CharacterClass(val rawValue: String) {
    Fighter("fighter"),
    Cleric("cleric"),
    Rogue("rogue"),
    Wizard("wizard"),
    Bard("bard");

    val title: String
        get() = rawValue.capitalize()

    companion object {
        val default = Fighter
        val list = values().toList()

        fun fromRawValue(rawValue: String): CharacterClass {
            return values().first { it.rawValue == rawValue }
        }
    }

    override fun toString(): String {
        return rawValue
    }
}